Name: Roam-Only in Device Menu
Version: 2.0.1-1
Author: Jason Robitaille
Description: Adds a roam-only toggle to the device menu  (MIT license)

--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
@@ -168,3 +168,4 @@
 		this.controller.get('ime_pref').addEventListener(Mojo.Event.tap,this.handleIMELaunch.bindAsEventListener(this));
+		this.controller.get('dm_roam').addEventListener(Mojo.Event.tap, this.toggleRoam.bindAsEventListener(this));
 
 		//Close the App because this is a first launch.
@@ -302,6 +303,30 @@
 		this.btdeviceList.clear();		
 	},
 	
+	toggleRoam: function(event) {
+		if(this.roamStatus=="roamonly") {
+			this.roamStatus="any";
+		} else {
+			this.roamStatus="roamonly";
+		}
+		this.controller.serviceRequest('palm://com.palm.telephony', {
+			method: 'roamModeSet',
+			parameters: {
+				mode: this.roamStatus,
+				client: Mojo.appName
+			}
+		});
+		this.setRoamLabel();
+	},
+
+	setRoamLabel: function() {
+		if(this.roamStatus=="roamonly") {
+			this.controller.get('roamStatus').innerText = "On";
+		} else {
+			this.controller.get('roamStatus').innerText = "Off";
+		}
+	},
+	
 	//Response from Preferences
 	gotNumOfProfiles: function(payload) {
 		Mojo.Log.info("Num Of Profile "+ Object.toJSON(payload));
@@ -1437,2 +1462,10 @@
 			this.isVisible = true;
+			this.controller.serviceRequest('palm://com.palm.telephony', {
+				method: 'roamModeQuery',
+				parameters: {},
+				onSuccess: function(response) {
+					this.roamStatus = response.extended.mode;
+					this.setRoamLabel();
+				}.bind(this)
+			});
 			var date = new Date();		
--- .orig/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
+++ /usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
@@ -89,6 +89,15 @@
 					
 				</div>	
 				
+				<div class="palm-section-divider"></div>
+
+				<div class="palm-row" id="dm_roam" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
+					<div class="title truncating-text">
+						<div class="label right" id="roamStatus">&nbsp;</div>
+						<span x-mojo-loc=''>Roam Only</span>
+					</div>
+				</div></div>
+
 				<div class="palm-section-divider"></div>
 				
 				<div id="dm_airplanemode" class="palm-row last menu-end" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
